src/main-process/atom-application.coffee

```diff
@@ -37,7 +37,7 @@ class AtomApplication
         userNameSafe = new Buffer(process.env.USERNAME).toString('base64')
         options.socketPath = "\\\\.\\pipe\\atom-#{options.version}-#{userNameSafe}-#{process.arch}-sock"
```

Comment:
Supporting this on Windows will probably require adding a *normalized* and *escaped*  `ATOM_HOME` in the pipe name. Normalizing and escaping is necessary to avoid paths that are the same but point to the same directory and to avoid adding backslashes to the pipe path.